source/rules/no-unsafe-switchmap.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { stripIndent } from "common-tags";
+import decamelize from "decamelize";
+import { Rule } from "eslint";
+import esquery from "esquery";
+import * as es from "estree";
+import { isCallExpression, isIdentifier, isLiteral, typecheck } from "../utils";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Disallows unsafe switchMap usage in Effects and epics.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Unsafe switchMap usage in Effects and epics is forbidden."
+    },
+    schema: [
+      {
+        properties: {
+          allow: {
+            oneOf: [
+              { type: "string" },
+              { type: "array", items: { type: "string" } }
+            ]
+          },
+          disallow: {
+            oneOf: [
+              { type: "string" },
+              { type: "array", items: { type: "string" } }
+            ]
+          },
+          observable: {
+            oneOf: [
+              { type: "string" },
+              { type: "array", items: { type: "string" } }
+            ]
+          }
+        },
+        type: "object",
+        description: stripIndent`
+          An optional object with optional \`allow\`, \`disallow\` and \`observable\` properties.
+          The properties can be specified as regular expression strings or as arrays of words.
+          The \`allow\` or \`disallow\` properties are mutually exclusive. Whether or not
+          \`switchMap\` is allowed will depend upon the matching of action types with \`allow\` or \`disallow\`.
+          The \`observable\` property is used to identify the action observables from which effects and epics are composed.
+        `
+      }
+    ]
+  },
+  create: context => {
+    const defaultObservable = "action(s|\\$)?";
+    const defaultDisallow = [
+      "add",
+      "create",
+      "delete",
+      "post",
+      "put",
+      "remove",
+      "set",
+      "update"
+    ];
+
+    function createRegExp(value: any): RegExp | null {
```

Comment:
TSLint has separate tests for this function - https://github.com/cartant/rxjs-tslint-rules/blob/master/source/rules/rxjsNoUnsafeSwitchmapRule-spec.ts 

How would you like to setup these tests in this repo?

Reply:
The tests are run using Mocha, so we should be able to do it the same way here. That is, stick them in a `tests/rules/no-unsafe-switchmap-spec.ts` file. Mocha will find the, as it'll match all `.ts` files under `tests`:

https://github.com/cartant/eslint-plugin-rxjs/blob/67a5957c5b65b95f6fbe3823eda5447fcdd6af4f/package.json#L75

Reply:
The rule has a default export, which is the rule so we can't export another element. I moved the function to the utils file, and create a `utils-spec.ts` file.